rds-snapper/main.go

```
@@ -0,0 +1,257 @@
+	input := &rds.DescribeDBSnapshotsInput{
+		SnapshotType:  aws.String("manual"),
+		IncludePublic: aws.Bool(true),
+		IncludeShared: aws.Bool(true),
+	}
+
+	result, err := svc.DescribeDBSnapshots(input)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	return result
+}
+
+func retreiveInstanceManualSnapshots(dBInstanceIdentifier string, svc *rds.RDS) *rds.DescribeDBSnapshotsOutput {
+	input := &rds.DescribeDBSnapshotsInput{
+		DBInstanceIdentifier: aws.String(dBInstanceIdentifier),
+		SnapshotType:         aws.String("manual"),
+		IncludePublic:        aws.Bool(true),
+		IncludeShared:        aws.Bool(true),
+	}
+
+	result, err := svc.DescribeDBSnapshots(input)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	return result
+}
+
+func cleanUpSnapshot(dBSnapshotIdentifier *string, svc *rds.RDS) {
+	input := &rds.DeleteDBSnapshotInput{
+		DBSnapshotIdentifier: aws.String(*dBSnapshotIdentifier),
+	}
+
+	result, err := svc.DeleteDBSnapshot(input)
+	if err != nil {
+		fmt.Println(err)
+	}
+	fmt.Println(result)
+}
+
+func maintainSnapshots(dBInstanceIdentifier string, svc *rds.RDS, limit int) {
+	input := retreiveInstanceManualSnapshots(dBInstanceIdentifier, svc)
+
+	if len(input.DBSnapshots) > limit {
+		sort.SliceStable(input.DBSnapshots, func(i int, j int) bool {
+			return input.DBSnapshots[i].SnapshotCreateTime.Before(*input.DBSnapshots[j].SnapshotCreateTime)
+		})
+		for index := 0; index < len(input.DBSnapshots)-limit; index++ {
+
+			cleanUpSnapshot(input.DBSnapshots[index].DBSnapshotIdentifier, svc)
+		}
+	}
+}
+
+func saveCsv(result *rds.DescribeDBSnapshotsOutput, filePath string) {
+	records := [][]string{}
+
+	for index := 0; index < len(result.DBSnapshots); index++ {
+		record := result.DBSnapshots[index]
+		records = append(records, []string{record.SnapshotCreateTime.String(), *record.DBInstanceIdentifier, *record.DBSnapshotIdentifier})
+	}
+
+	outfile, err := os.Create(filePath)
+	if err != nil {
+		log.Fatal("Unable to open output")
```

Not entirely familiar with error handling practices on golang yet, but I'd prefer to no randomly stop the program in the middle of a subroutine (as is implied in the `log.Fatal` call) - rather return the error instead and let the caller decide what to do.